### PR TITLE
[Silabs] Reduce code size for the Lighting-app

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Remove unused feature in the Lighting-app including 

- BLE features like scanner
- Fixed Label clusters
- READ_Client
- etc...

### CODE REVIEW

File: third_party/silabs/matter_support
Comment (andy31415) [inline third_party/silabs/matter_support:1]: Are the silabs changes also expected? The summary says "remove unused bits" however not silabs matter support files which seem to contain ICD defines and config re-ordering

Comment (jmartinez-silabs) [reply third_party/silabs/matter_support:1]: The changes in matter/si91x/siwx917/BRD4338A/autogen that you are pointing at are indeed changes from a commit before @jepenven-silabs lastest commit in that submodule. Those changes were not yet brought into usage here and are causing the CI build failure.

The changes are okay, but we must remove `sl_si91x_m4_ps.c` from the build.

**Review by jmartinez-silabs — APPROVED** [review]
Approved, but need to fix 917 build due to the unrelated changes brought up with the submodule update

File: third_party/silabs/matter_support
Comment (jepenven-silabs) [reply third_party/silabs/matter_support:1]: Resolved ish ? I'll let a Silabs WiFi expert fix the behaviour on monday, for now the build should be fixed.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**